esphome/components/nextion/display.py

```diff
@@ -172,9 +172,14 @@ async def to_code(config):
 
     cg.add(var.set_auto_wake_on_touch(config[CONF_AUTO_WAKE_ON_TOUCH]))
 
-    cg.add(var.set_exit_reparse_on_start(config[CONF_EXIT_REPARSE_ON_START]))
-
-    cg.add(var.set_skip_connection_handshake(config[CONF_SKIP_CONNECTION_HANDSHAKE]))
+    if CONF_EXIT_REPARSE_ON_START in config and config[CONF_EXIT_REPARSE_ON_START]:
+        cg.add_define("USE_NEXTION_CONFIG_EXIT_REPARSE_ON_START")
```

Comment:
default in config is `False` so no need to check
```suggestion
    if config[CONF_EXIT_REPARSE_ON_START]:
        cg.add_define("USE_NEXTION_CONFIG_EXIT_REPARSE_ON_START")
```

Reply:
But if someone explicitly adds it as `false` (even when not needed, as it is default), it will be detected in the config and flag that, which is not desired.